src/components/Calculator.js

```diff
@@ -1,71 +1,83 @@
+import React, { useState } from 'react';
+import calculate from '../logic/calculate';
 import '../styles/Calculator.css';
 
 function Calculator() {
+  const [calculatorData, setCalculatorData] = useState({
+    total: null,
+    next: null,
+    operation: null,
+  });
+
+  const handleClick = (event) => {
+    const buttonName = event.target.value;
+    const newData = calculate(calculatorData, buttonName);
+    setCalculatorData(newData);
+  };
+
+  const displayValue = calculatorData.next || calculatorData.total || '0';
+
   return (
     <main className="calculator">
       <section className="input">
-        <input className="enter" type="text" readOnly />
+        <input className="calcDisplay" value={displayValue} type="text" readOnly />
       </section>
       <section className="buttons grid-container">
-        <button className="grid-item item1" value="AC" type="button">
+        <button className="grid-item item1" onClick={handleClick} value="AC" type="button">
           AC
         </button>
-        <button className="grid-item item2" value="+/-" type="button">
+        <button className="grid-item item2" onClick={handleClick} value="+/-" type="button">
           +/-
         </button>
-        <button className="grid-item item3" value="%" type="button">
+        <button className="grid-item item3" onClick={handleClick} value="%" type="button">
           %
         </button>
-        <button className="grid-item item4" value="÷" type="button">
+        <button className="grid-item item4" onClick={handleClick} value="÷" type="button">
           ÷
         </button>
-
-        <button className="grid-item item5" value="7" type="button">
+        <button className="grid-item item5" onClick={handleClick} value="7" type="button">
           7
         </button>
-        <button className="grid-item item6" value="8" type="button">
+        <button className="grid-item item6" onClick={handleClick} value="8" type="button">
           8
         </button>
-        <button className="grid-item item7" value="9" type="button">
+        <button className="grid-item item7" onClick={handleClick} value="9" type="button">
           9
         </button>
-        <button className="grid-item item8" value="x" type="button">
+        <button className="grid-item item8" onClick={handleClick} value="x" type="button">
           x
         </button>
-
-        <button className="grid-item item9" value="4" type="button">
+        <button className="grid-item item9" onClick={handleClick} value="4" type="button">
           4
         </button>
-        <button className="grid-item item10" value="5" type="button">
+        <button className="grid-item item10" onClick={handleClick} value="5" type="button">
```

Comment:
**OPTIONAL**

- [ ] It can be nice to great the buttons as a separate component, then pass the values as props. This will reduce the dublication. Also, you can use a loop to display the buttons.